src/main/scala/uncore/tilelink2/Crossing.scala

```diff
@@ -51,12 +59,19 @@ class TLAsyncCrossingSink(depth: Int = 8, sync: Int = 3) extends LazyModule
     }
 
     ((io.in zip io.out) zip (node.edgesIn zip node.edgesOut)) foreach { case ((in, out), (edgeIn, edgeOut)) =>
+      val reset_n = in.a.source_reset_n
       val bce = edgeOut.manager.anySupportAcquire && edgeOut.client.anySupportProbe
 
       out.a <> FromAsyncBundle(in.a, sync)
       in.d <> ToAsyncBundle(out.d, depth, sync)
 
+      assert (!out.d.valid || reset_n, "D channel respose sent to missing client")
+
       if (bce) {
+        val up = RegInit(Bool(false))
+        when (reset_n) { up := Bool(true) }
```

Comment:
I suppose one could rename it to sink_reset_n and source_reset_n. I'll make a change to that effect now.